src/libaktualizr/primary/sotauptaneclient.h

```diff
@@ -76,11 +79,11 @@ class SotaUptaneClient {
   FRIEND_TEST(Uptane, offlineIteration);
   FRIEND_TEST(Uptane, IgnoreUnknownUpdate);
   FRIEND_TEST(Uptane, kRejectAllTest);
-  FRIEND_TEST(Uptane, Vector);  // Note hacky name (see uptane_vector_tests.cc)
   FRIEND_TEST(UptaneCI, ProvisionAndPutManifest);
   FRIEND_TEST(UptaneCI, CheckKeys);
   FRIEND_TEST(UptaneKey, Check);  // Note hacky name
   FRIEND_TEST(UptaneNetwork, DownloadFailure);
+  FRIEND_TEST(UptaneVector, Test);
```

Comment:
At first sight, looks like too generic name.

Reply:
Yes, I struggled with the naming here. The problem is that the test case is generic by definition, because it is dynamically reused to run each individual Uptane vector test. If you look at `tests/uptane_vector_tests.cc` you'll note that I have to name a class with the first part of that, and I think naming the class `Uptane` (so that the test case name is "VectorTest" or something) is too generic and potentially conflicts with the "Uptane" namespace. If you have a suggestion, I'll take it.

Reply:
Yeah, I understand now better why it's called Test and don't have better name though :), perhaps it's enough to have UptaneVector as a namespace. 
Another idea, that I think, I already mentioned is to add python binding to libaktualizr (e.g. https://www.boost.org/doc/libs/1_68_0/libs/python/doc/html/tutorial/index.html) which can be useful for two things:

1) Integration & Functional tests (e.g. the uptane vector tests) written in Python so each test business logic is encapsulated in one place (a single test) instead of being spread across different files written on different languages.

2) For community/customers/app developers who would prefer to develop apps on top of libaktualizr in Python. 

This is just IMHO and I am not sure if it's feasible and there is any business & effort justification to do it all.

Reply:
> Another idea, that I think, I already mentioned is to add python binding to libaktualizr

That is interesting, but probably not worth the investment at present. Generally, when we use Python for testing, we are calling an executable directly and actually do not want to reference classes directly. If we were to want to do that, why not just use C++?